### PR TITLE
bugfix for dot of Hermitian{noncommutative}

### PR DESCRIPTION
**Pull request by stevengj**
Fixes a bug identified by @andreasnoack in #52318, where `dot(A,B)` was giving incorrect result for `Hermitian` matrices over fields like quaternions where `conj(a)*conj(b) ≠ conj(a*b)`

(The bug was introduced in #32827, which was part of Julia 1.4.)

### CODE REVIEW

**Review by andreasnoack — APPROVED** [review]

**Comment by stevengj** [comment]
CI is giving:
```
/Users/julia/.julia/scratchspaces/a66863c6-20e8-4ff4-8a62-49f30b1f605e/agent-cache/default-macmini-x64-5.0/build/default-macmini-x64-5-0/julialang/julia-master/julia-b0777064fd/share/julia/test/testdefs.jl:24
  Got exception outside of a @test
  LoadError: UndefVarError: `BASE_TEST_PATH` not defined in `Main.Test74Main_LinearAlgebra_symmetric.TestSymmetric`
  Stacktrace:
```

File: stdlib/LinearAlgebra/test/symmetric.jl [hunk 1]
```diff
@@ -4,6 +4,9 @@ module TestSymmetric
 
 using Test, LinearAlgebra, Random
 
+isdefined(Main, :Quaternions) || @eval Main include(joinpath($(BASE_TEST_PATH), "testhelpers", "Quaternions.jl"))
```
Comment (stevengj) [inline stdlib/LinearAlgebra/test/symmetric.jl:7]:
```suggestion
const BASE_TEST_PATH = joinpath(Sys.BINDIR, "..", "share", "julia", "test")

isdefined(Main, :Quaternions) || @eval Main include(joinpath($(BASE_TEST_PATH), "testhelpers", "Quaternions.jl"))
```

File: stdlib/LinearAlgebra/test/symmetric.jl [hunk 2]
```diff
@@ -462,6 +467,14 @@ end
     end
 end
 
+# bug identified in PR #52318: dot products of quaternionic Hermitian matrices,
+# or any number type where conj(a)*conj(b) ≠ conj(a*b):
+@testset "dot Hermitian quaternion #52318" begin
+    A, B = [Quaternion.(randn(3,3), randn(3, 3), randn(3, 3), randn(3,3)) |> t -> t + t' for i in 1:2]
+    @test A == Hermitian(A) && B == Hermitian(B)
+    @test dot(A,B) ≈ dot(Hermitian(A),Hermitian(B))
```
Comment (dkarrasch) [inline stdlib/LinearAlgebra/test/symmetric.jl:475]: I'd like to propose to add a corresponding symmetric test. It is currently passing, but I'm not sure if the explicit realification is still valid without also restricting `Symmetric` to commutative eltypes.

```suggestion
    A, B = [Quaternion.(randn(3,3), randn(3, 3), randn(3, 3), randn(3,3)) |> t -> t + t' for i in 1:2]
    @test A == Hermitian(A) && B == Hermitian(B)
    @test dot(A, B) ≈ dot(Hermitian(A), Hermitian(B))
    A, B = [Quaternion.(randn(3,3), randn(3, 3), randn(3, 3), randn(3,3)) |> t -> t + transpose(t) for i in 1:2]
    @test A == Symmetric(A) && B == Symmetric(B)
    @test dot(A, B) ≈ dot(Symmetric(A), Symmetric(B))
```

Comment (dkarrasch) [reply stdlib/LinearAlgebra/test/symmetric.jl:475]: Indeed, I noticed that later.

**Pull request merged**